intervaltree/intervaltree.py

```diff
@@ -437,6 +437,17 @@ def intersection_update(self, other):
             if iv not in other:
                 self.remove(iv)
 
+    def overlap_intervals(self,other):
+        """
+        Returns a new IntervalTree consisting of intervals representing the
+        regions overlapped by at least one interval in both of self and other.
+        """
+        splits = (self | other)
+        splits.split_overlaps()
```

Comment:
Hmm, I am sure you are right ... but it goes beyond my expertise to implement it using the internals, with the time I have available in the near future. I am happy for you not to merge this change if the performance worries you - I only offered it because I am using it very often myself due to porting a lot of code over from other libraries which have this function, and in practice it works for me well enough that I felt it might be useful to contribute.